File: PaymentService/src/main/kotlin/it/polito/wa2/lab5/g09/paymentservice/entities/Transaction.kt
package it.polito.wa2.lab5.g09.paymentservice.entities
import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.sql.Timestamp

@Document(collection = "user_transaction")
data class Transaction(
    @Id
    @Indexed
    val transactionId: ObjectId = ObjectId.get(),
    val amount: Double,
    val customerUsername: String,
    val orderId: ObjectId,
    val date: Timestamp = Timestamp(System.currentTimeMillis()),
    val isConfirmed: Boolean?
)

File: client/courseprojectg09/src/model/Transaction.js
class Transaction {
    constructor(transactionId, amount, date, isConfirmed, orderId, customerUsername) {
        this.transactionId = transactionId;
        this.amount = amount;
        this.date = date;
        this.isConfirmed = isConfirmed;
        this.orderId = orderId;
        this.customerUsername = customerUsername;
    }

    static from(json) {
        let newJson = {transactionId: json.transactionId, amount: `€ ${json.amount}`, date: convertDate(json.date), isConfirmed: (json.isConfirmed ? 'ACCEPTED' : 'REJECTED') , orderId: json.orderId, customerUsername: json.customerUsername}
        const transaction = new Transaction();
        Object.assign(transaction, newJson);
        return transaction;
    }
}

function convertDate(date){
    let d =  new Date(date).toLocaleDateString("it-IT")
    let t = new Date(date).toLocaleTimeString("it-IT")
    return `${d} - ${t}`
}

export default Transaction;